src/encoded/static/components/glossary.js

```diff
@@ -1,24 +1,77 @@
+                    <span className="glossary-field">{props.glossaryEntry.file_formats}</span>
+                </p>
+            : null}
+            {props.glossaryEntry.additional_information ?
+                <p className="glossary-extra-info">
+                    <span className="glossary-label">Additional information: </span>
+                    {props.glossaryEntry.additional_information.map(href =>
+                        <span className="glossary-field" key={href.url}>
+                            <a href={href.url}>{href.title}</a>
+                        </span>
+                    )}
+                </p>
+            : null}
+        </div>
+    );
+};
+
+GlossaryTerm.propTypes = {
+    glossaryEntry: PropTypes.object.isRequired,
+};
+
 const Glossary = (props) => {
     const { context } = props;
     const glossary = require('./glossary/glossary.json');
+    const glossaryBySection = _(glossary).groupBy(entry => entry.section || 'General terms');
 
     return (
         <div className="layout">
             <div className="layout__block layout__block--100">
-                <div className="ricktextblock block" data-pos="0,0,0">
+                <div className="richtextblock block" data-pos="0,0,0">
                     <center>
                         <h1>{context.title}</h1>
                     </center>
-                    {glossary.map(g => (
-                        <div key={g.term} className="glossary-element">
-                            <h4 className="glossary-term">{g.term}</h4>
-                            <p className="glossary-definition">{g.definition}</p>
+                    <h4 className="directory">
+                        {Object.keys(glossaryBySection).map((section, sectionIdx) =>
+                            <React.Fragment key={`${section}-link`}>
+                                <a href={`#${section.toLowerCase().split(' ').join('-')}`}>{section}</a>
+                                {(sectionIdx < Object.keys(glossaryBySection).length - 1) ?
+                                    <span> | </span>
+                                : null}
+                            </React.Fragment>
+                        )}
+                    </h4>
+                    {Object.keys(glossaryBySection).map(section =>
+                        <div className="glossary-block" key={section} id={section}>
+                            <h2 id={`${section.toLowerCase().split(' ').join('-')}`}>{section}</h2>
+                            {(section === undefined) ?
+                                <h4>{section}</h4>
+                            : null}
+                            {glossaryBySection[section]
+                                .sort((a, b) => {
+                                    if (a.term.toLowerCase() < b.term.toLowerCase()) { return -1; }
+                                    if (a.term.toLowerCase() > b.term.toLowerCase()) { return 1; }
+                                    return 0;
```

Comment:
This is a good idea. I don't think IE-11 supports it but that is no big deal. I am forward to August when all we need to worry about are evergreen browsers.